addons/mail/tests/common.py

```diff
@@ -349,19 +349,25 @@ def _find_mail_mail_wrecord(self, record, status=None, mail_message=None, author
             )
         return mail
 
-    def _find_sent_email(self, email_from, emails_to, subject=None):
-        """ Find an outgoing email based on from / to and optional subject when
-        having a conflict.
+    def _find_sent_email(self, email_from, emails_to, subject=None, body=None, attachment_names=None):
+        """ Find an outgoing email based on from / to and optional subject, body
+        and attachment names when having conflicts.
 
         :return sent_email: an outgoing email generated during the mock;
         """
         sent_emails = [
             mail for mail in self._mails
-            if set(mail['email_to']) == set(emails_to) and mail['email_from'] == email_from
+            if (set(email_normalize(email) for email in mail['email_to']) == set(email_normalize(email) for email in emails_to)
+                and mail['email_from'] == email_from)
         ]
-        if len(sent_emails) > 1 and subject:
+        if len(sent_emails) > 1:
             # try to better filter
-            sent_email = next((mail for mail in sent_emails if mail['subject'] == subject), False)
+            sent_email = next((mail for mail in sent_emails
+                               if (subject is None or mail['subject'] == subject)
+                               and (body is None or mail['body'] == body)
+                               and (attachment_names is None
+                                    or set(attachment_names) <= set(attachment[0] for attachment in mail['attachments']))
```

Comment:
Can't we find the right set of attachments instead of "contained in", to avoid false positives ?

Reply:
Isn't it possible to either fix those tests ? Do we have a lot of them ?

Reply:
:thinking: I'll see, I'm guessing now it ought to have been something else because I can't see why I wouldn't have just fixed them in that case..